rwengine/src/objects/PickupObject.cpp

```diff
@@ -225,3 +227,142 @@ void PickupObject::setEnabled(bool enabled) {
+            break;
+        case 174: /* Molotov */
+            totalRounds = 5;
+            break;
+        case 181: /* Flame thrower */
+            totalRounds = 25;
+            break;
+        case 171: /* AK */
+            totalRounds = 150;
+            break;
+        case 180: /* M16 */
+            totalRounds = 300;
+            break;
+        case 177: /* Sniper Rifle */
+            totalRounds = 25;
+            break;
+    }
+
+    if (getPickupType() == OnStreet || getPickupType() == OnStreetSlow) {
+        totalRounds /= 5;
+    }
+
+    character->addToInventory(item->inventorySlot, totalRounds);
+
+    return true;
+}
+
+ScriptedPickup::ScriptedPickup(GameWorld* world, const glm::vec3& position,
+                               BaseModelInfo* modelinfo, PickupType type)
+    : PickupObject(world, position, modelinfo, type) {
+}
+
+bool ScriptedPickup::onCharacterTouch(CharacterObject* character) {
+    if (!character->isPlayer()) {
+        return false;
+    }
+
+    return true;
+}
+
+HealthPickup::HealthPickup(GameWorld* world, const glm::vec3& position,
+                           BaseModelInfo* modelinfo, PickupType type)
+    : PickupObject(world, position, modelinfo, type) {
+}
+
+bool HealthPickup::onCharacterTouch(CharacterObject* character) {
+    if (!character->isPlayer()) {
+        return false;
+    }
+
+    if (character->getCurrentState().health >= 100.f) {
+        return false;
+    }
+
+    character->getCurrentState().health = 100.f;
+
+    return true;
+}
+
+ArmourPickup::ArmourPickup(GameWorld* world, const glm::vec3& position,
+                           BaseModelInfo* modelinfo, PickupType type)
+    : PickupObject(world, position, modelinfo, type) {
+}
+
+bool ArmourPickup::onCharacterTouch(CharacterObject* character) {
+    if (!character->isPlayer()) {
+        return false;
+    }
+
+    if (character->getCurrentState().armour >= 100.f) {
+        return false;
+    }
+
+    character->getCurrentState().armour = 100.f;
+
+    return true;
+}
+
+CollectablePickup::CollectablePickup(GameWorld* world,
+                                     const glm::vec3& position,
+                                     BaseModelInfo* modelinfo, PickupType type)
+    : PickupObject(world, position, modelinfo, type) {
+}
+
+bool CollectablePickup::onCharacterTouch(CharacterObject* character) {
+    if (!character->isPlayer()) {
+        return false;
+    }
+
+    GameState* state = character->engine->state;
+
+    if (state->playerInfo.hiddenPackagesCollected ==
+        state->playerInfo.hiddenPackageCount) {
+        state->playerInfo.money += 1000000;
+
+        // @todo display "heres little something"
+    } else {
+        state->playerInfo.hiddenPackagesCollected++;
+        state->playerInfo.money += 1000;
+
+        // @todo display "hidden packages found x of y"
+        const auto gxtEntry = "CO_ONE";
+
+        auto str = ScreenText::format(
+            character->engine->data->texts.text(gxtEntry),
```

Comment:
There should be an engine member here already, without needing to access through the player.